### PR TITLE
Install renderer, add test cases to the operate,calculate and home component

### PR DESCRIPTION
**Pull request by Fadahunsiseyi**
-Add test renderer.
-Add tests to the operate.js file.
-Add tests to the calculate.js file.
-Add tests to the home component.

### CODE REVIEW

File: src/logic/calculate.test.js [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import calculate from './calculate';
+
+it('should return an object', () => {
+  const calculateTest = calculate({ total: 0, next: null, operation: null }, 'AC');
+  expect(calculateTest).toStrictEqual({ total: 0, next: null, operation: null });
+});
```
Comment (omarsalem7) [inline src/logic/calculate.test.js]: - [ ] Kindly make sure that all the functionalities & scenarios of the `calculate.js` (numbers, operations: +, - , * , / , % )  should be covered using Jest. And add a good description for every test.

**Review by omarsalem7 — CHANGES_REQUESTED** [review]
Hi @Fadahunsiseyi, @sajjadrajaee 😊, 

Good job so far! 👏
You have done well reaching this point of the milestone, but there are some issues that you still need to work on to go to the next project but you are almost there! 

you-are-near

## Highlights
- Descriptive pull request ✔️
- Good usage of Jest snapshots for testing all the components. ✔️

## Required Changes ♻️
_Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/calculate.test.js (hunk 1 above)
Comment (Fadahunsiseyi) [reply src/logic/calculate.test.js]: Hi @omarsalem7 , please I don't understand what this review says. Please can you explain?

File: src/logic/calculate.test.js [hunk 2]
```diff
@@ -1,6 +1,16 @@
...
+      next: null,
+      operation: null,
+    }, 'AC');
+    expect(result).toEqual({
+      total: 0,
+      next: null,
+      operation: null,
+    });
+  });
 });
```
Comment (HAMMAS-SALEEM) [inline src/logic/calculate.test.js]: - [ ] You are just testing your AC button. Please add unit tests in which total, next, operation values will change. :blush: as previous reviewer asked you to do it. If it is difficult for you to understand then reach out to me on slack by searching my name **Hammas** in **full time student** channel :+1:

**Review by HAMMAS-SALEEM — CHANGES_REQUESTED** [review]
# Required Changes :recycle:
almost-there

### Hey @Fadahunsiseyi. @sajjadrajaee :wave:

### Highlights :1st_place_medal:
 
:heavy_check_mark: Following Linters
:heavy_check_mark: Descriptive PR Summary

Good job on implementing some of the changes requested by previous reviewer, there are some issues that you still need to work on to go to the next project but you are almost there! :blush:

# Optional Suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @HAMMAS-SALEEM in your question so I can receive the notification.

_**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_

------------

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/logic/calculate.test.js (hunk 2 above)
Comment (Fadahunsiseyi) [reply src/logic/calculate.test.js]: Thank you so much for your time @HAMMAS-SALEEM

File: src/logic/calculate.test.js (hunk 1 above)
Comment (Fadahunsiseyi) [reply src/logic/calculate.test.js]: Thank you for taking your time to reach out @omarsalem7

**Review by mahbubul14 — APPROVED** [review]
# Project Approved :trophy: :tada: 

Hi @Team, 

Your project is complete!  You have done it exceptionally well. it's time to merge it :100: 
Congratulations! 🎉 

### To Highlight :+1: 

:heavy_check_mark: No linter error.
:heavy_check_mark: Descriptive readme file.
:heavy_check_mark: Well implemented requested changes.

### Optional suggestions :arrow_down_small: 

* N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also connect with me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**